manifest_staging/charts/secrets-store-csi-driver/templates/crds-delete-hook.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ template "sscd.fullname" . }}-delete-crds
+  namespace: {{ .Release.Namespace }}
+{{ include "sscd.labels" . | indent 2 }}
+  annotations:
+    helm.sh/hook: post-delete
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+rules:
+  - apiGroups: ["apiextensions.k8s.io"]
+    resources: ["customresourcedefinitions"]
+    verbs: ["delete"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ template "sscd.fullname" . }}-delete-crds
+  namespace: {{ .Release.Namespace }}
+{{ include "sscd.labels" . | indent 2 }}
+  annotations:
+    helm.sh/hook: post-delete
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+subjects:
+  - kind: ServiceAccount
+    name: {{ template "sscd.fullname" . }}-delete-crds
+    namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: {{ template "sscd.fullname" . }}-delete-crds
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "sscd.fullname" . }}-delete-crds
+  namespace: {{ .Release.Namespace }}
+{{ include "sscd.labels" . | indent 2 }}
+  annotations:
+    helm.sh/hook: post-delete
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ template "sscd.fullname" . }}-delete-crds
+  namespace: {{ .Release.Namespace }}
+{{ include "sscd.labels" . | indent 2 }}
+  annotations:
+    helm.sh/hook: post-delete
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+spec:
+  serviceAccountName: {{ template "sscd.fullname" . }}-delete-crds
+  restartPolicy: Never
+  containers:
+    - name: delete-crds
+      image: "{{ .Values.linux.crds.image.repository }}:{{ .Values.linux.crds.image.tag }}"
+      args:
+      imagePullPolicy: {{ .Values.linux.registrarImage.pullPolicy }}
+      command:
+      - sh
+      - -c
+      - >
+       kubectl delete crd secretproviderclasses.secrets-store.csi.x-k8s.io --ignore-not-found;
```

Comment:
Why are we deleting the crds? This can lead to loss of data. 

By having crds in `template` dir, helm would delete the crds. We're moving the crds to `crds` dir to prevent that and also conform with helm3 recommendations for crds.

Reply:
helm uninstall does not delete crds from `crds` dir. At the time of helm install, it will apply configuration given in crds only if resources with same name is not already present.

Reply:
That's right! And we also don't want to delete the crd. So do we need this hook for deletion?

Reply:
Cool. Removed it